File: api/app/src/main/kotlin/com/scribblefit/api/features/exercises/ExerciseDto.kt
package com.scribblefit.api.features.exercises

import kotlinx.serialization.Serializable

@Serializable
data class ExerciseDto(
    val id: String,
    val canonicalName: String,
    val muscleGroup: String,
    val aliases: List<String>
)

@Serializable
data class ExerciseResponse(
    val exercises: List<ExerciseDto>
)

File: apps/ios/ScribbleFit/Sources/Core/Network/Models/ExerciseDto.swift
import Foundation

public struct ExerciseDto: Codable, Sendable {
    public let id: String
    public let canonicalName: String
    public let muscleGroup: String
    public let aliases: [String]
    
    public init(id: String, canonicalName: String, muscleGroup: String, aliases: [String]) {
        self.id = id
        self.canonicalName = canonicalName
        self.muscleGroup = muscleGroup
        self.aliases = aliases
    }
}

public struct ExerciseResponse: Codable, Sendable {
    public let exercises: [ExerciseDto]
    
    public init(exercises: [ExerciseDto]) {
        self.exercises = exercises
    }
}
